packages/plugin-figma-blade-coverage/src/main.ts

```diff
@@ -0,0 +1,429 @@
+      'nonBladeTextStyles#45789:5': nonBladeTextStyles.toString().padStart(2, '0'),
+      'nonBladeColorStyles#45789:6': nonBladeColorStyles.toString().padStart(2, '0'),
+    });
+
+    const detachedCoverageCard = coverageCardInstance.detachInstance();
+    traverseNode(detachedCoverageCard, (traversedNode) => {
+      if (traversedNode.type === 'TEXT') {
+        if (['bladeCoverageType', 'bladeCoverage'].includes(traversedNode.name)) {
+          traversedNode.setRangeFillStyleId(
+            0,
+            traversedNode.characters.length,
+            coverageColorIntent,
+          );
+        }
+      } else if (traversedNode.type === 'RECTANGLE' && traversedNode.name === 'Progress') {
+        traversedNode.resizeWithoutConstraints(bladeCoverageProgress, 4);
+        traversedNode.fillStyleId = coverageColorIntent;
+      }
+    });
+    detachedCoverageCard.visible = true;
+    bladeCoverageCards.push(detachedCoverageCard);
+  } catch (error: unknown) {
+    console.error(error);
+    figma.notify('⚠️ Error in rendering coverage card. Please try again', { error: true });
+    figma.closePlugin();
+  }
+};
+
+const calculateCoverage = (node: SceneNode): CoverageMetrics | null => {
+  let bladeComponents = 0;
+  let bladeTextStyles = 0;
+  let bladeColorStyles = 0;
+  let nonBladeComponents = 0;
+  let nonBladeTextStyles = 0;
+  let nonBladeColorStyles = 0;
+  let totalLayers = 0;
+
+  try {
+    if (getParentNode(node)?.type === 'PAGE' && !MAIN_FRAME_NODES.includes(node.type)) {
+      // if there are non-frame nodes as direct children of a page, ignore them
+      return null;
+    }
+
+    traverseNode(
+      node,
+      (traversedNode) => {
+        if (!traversedNode.visible) {
+          return;
+        }
+
+        if (
+          traversedNode.type === 'INSTANCE' &&
+          (BLADE_COMPONENT_IDS.includes(
+            (traversedNode.mainComponent?.parent as ComponentSetNode)?.key ?? '',
+          ) ||
+            BLADE_COMPONENT_IDS.includes(traversedNode.mainComponent?.key ?? ''))
+        ) {
+          bladeComponents++;
+        } else if (traversedNode.type === 'INSTANCE') {
+          nonBladeComponents++;
+          highlightNonBladeNode(traversedNode);
+        } else if (traversedNode.type === 'TEXT') {
+          // check if the text is using Blade's text styles
+          let isMixedTextStyleOfBlade = false;
+          let traversedNodeTextStyleId = '';
+          let isMixedColorStyleOfBlade = false;
+          let traversedNodeColorStyleId = '';
+
+          /** the textSyleId can have figma.mixed. so in that case we need to go character by character
+           * and do getRangeTextStyleId(charIndex,charIndex+1) instead of textStyleId
+           * */
+          if (traversedNode?.textStyleId === figma.mixed) {
+            isMixedTextStyleOfBlade = traversedNode.characters
+              .split('')
+              .every((character, index) => {
```

Comment:
Do we have to go to every character? 
Can we do `getRangeTextStyleId(0, traversedNode.character.length)`?

Reply:
if your text style has multiple styles then figma will return `figma.mixed` instead of a style id but it doesn't tells me what is different in that. So if you have something like this 

Qty: **2** _units_

in this figma will return the style as mixed hence i need to do character by character to check if each character has the color style/text style from blade 😢

Reply:
I see, makes sense.